### PR TITLE
V5 ga 117 generate certificate

### PR DESCRIPTION
**Pull request by SrdjanReblaze**
Screenshot from 2023-04-27 11-02-09
Screenshot from 2023-04-27 11-01-55
Screenshot from 2023-04-27 11-01-39

https://reblaze.atlassian.net/jira/software/projects/V5GA/boards/39?selectedIssue=V5GA-117

### CODE REVIEW

File: src/components/SslEditCertificate.vue [hunk 1]
```diff
@@ -125,24 +108,21 @@
...
-            </button>
+          <div class="column is-3 has-text-right columns">
+            <form-button label="Cancel"
+                         classes="column"
+                         variant="secondary"
+                         @click="resetEditModal"
+                         v-if="!isLoading"/>
+            <form-button @click="saveChanges"
+                         label="Save"
+                         :disabled="isSaveEditDisabled"
```
Comment (Aviv-Galmidi) [inline src/components/SslEditCertificate.vue:124]: Button is not disabled when it should be

File: src/components/SslGenerateCertificate.vue [hunk 2]
```diff
@@ -157,7 +156,7 @@ export default defineComponent({
       password: '',
       passwordMessage: '',
       isExtracting: false,
-      isManualInput: true,
+      isManualInput: 'manual',
```
Comment (Aviv-Galmidi) [inline src/components/SslGenerateCertificate.vue:159]: Let's change the name to `inputType` and the type to `'manual' | 'extract'`

File: src/views/premium-security/DynamicRulesEditor.vue
Comment (Aviv-Galmidi) [inline src/views/premium-security/DynamicRulesEditor.vue:1]: Added by accident?

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]
1. Tests need to pass

2. Modals weren't converted to the new modal component

3. Divider line `<hr>` wasn't converted to the new divider line component

4. Buttons are clamped together for new and edit modals

Screen Shot 2023-05-01 at 9 42 22

5. Buttons are not aligned in delete modal

Screen Shot 2023-05-01 at 9 47 04

6. Console errors when trying to save certificate after editing the attached apps

Screen Shot 2023-05-01 at 10 09 52

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]
Comments from previous review that were not fixed:

1. Modals weren't converted to the new modal component

2. The selected apps do not show up as selected by default
https://github.com/reblaze/ui/pull/262/#discussion_r1181426690

3. The dropdown doesn't show up when there are no items to display (placeholder expected)
https://github.com/reblaze/ui/pull/262/#discussion_r1181427920

4. Let's change the name to inputType and the type to 'manual' | 'extract'
https://github.com/reblaze/ui/pull/262/#discussion_r1181432370

New items:

5. vue.config.js needs to be reverted to the original code

6. Warning when clicking the radio button "Extract pfx file" in new form
Screen Shot 2023-05-18 at 13 00 19

7. Error when uploading pfx file
Screen Shot 2023-05-18 at 13 01 09

8. Extract password and button not aligned
Screen Shot 2023-05-18 at 13 01 38

**Comment by SrdjanReblaze** [comment]
all comments fixed, please check this is most advanced conversion because of different response of multiselect dropdown

**Comment by Aviv-Galmidi** [comment]
Comments from previous review that were not fixed:

* Modals weren't converted to the new modal component

* The dropdown doesn't show up when there are no items to display (placeholder expected)
`https://github.com/reblaze/ui/pull/262/#discussion_r1181427920`

* Let's change the name to inputType and the type to 'manual' | 'extract'
`https://github.com/reblaze/ui/pull/262/#discussion_r1181432370`

* Warning when clicking the radio button "Extract pfx file" in new form

**Comment by SrdjanReblaze** [comment]
> Comments from previous review that were not fixed:
> 
> * Modals weren't converted to the new modal component
> * The dropdown doesn't show up when there are no items to display (placeholder expected)
>   `https://github.com/reblaze/ui/pull/262/#discussion_r1181427920`
> * Let's change the name to inputType and the type to 'manual' | 'extract'
>   `https://github.com/reblaze/ui/pull/262/#discussion_r1181432370`
> * Warning when clicking the radio button "Extract pfx file" in new form
> 
> 

fixed, please check about inputType and no items dropdown, it looks good to me, modal conversion done for generate, edit and  delete too

**Comment by Aviv-Galmidi** [comment]
Added as part of the branch `ux-upgrade-next`

**Pull request closed without merging**